### DISCUSSION TITLE
Revisiting nbsphinx vs. MyST-NB for roadmap plans

### DISCUSSION THREAD

**Discussion by GlobalMin** [executablebooks/meta#1035] [General]
Hi all. 👋 
### Background
nbsphinx and MyST-NB have been compared and discussed at length so [I'll link to this thread](https://github.com/spatialaudio/nbsphinx/issues/420) as a very relevant place even though it was ~3 years ago. RTDs has a page in [their own official docs](https://docs.readthedocs.io/en/stable/guides/jupyter.html#background) referencing both packages and some of the differences. So I'm not looking to rehash the same topic again and overall it seems that the functionality is very similar but there are gaps that are largely due to very different ways of rendering the .md.
### Main topics to the community
The nbsphinx research I've done recently led to some topics I want to put to the community:
* **Dependency constraints**: nbsphinx seems to be compatible with `MyST-Parser>=1.0`. Any plans to ultimately align the latest versions of MyST-Parser and MyST-NB/JB so the latter two can leverage the latest MyST-Parser updates?
*  **Customizing style with pygments:** Very easy to change the pygment style using the standard Sphinx option in conf.py. I have not been able to change the syntax highlighting or style with this option in MyST-NB or JupyterBook, but if it is possible I would love to know how. I am aware of the ANSI color note in EB's docs but I haven't seen this used elsewhere or don't know how to go about leveraging this method to modify syntax highlighting style. If an example exists, I would appreciate a link to it and if not I would vote for adding this to help users. The pygments option in Sphinx provides quick options though beyond the text highlighting though, which I find can add a noticeable style difference to the whole page.   </br>
The one example I know of that doesn't look anything like the sphinx-book-theme but is run on JupyterBook is the [quantecon-book-theme](https://github.com/QuantEcon/quantecon-book-theme). It's an amazing theme and shows what is possible in terms of building something from the ground up but it's not a template for quick, small edits to modify style. Put another way, most of the docs built on JupyterBook seem to stick with the default style or have minimal changes. I like the aesthetic of sphinx-book-theme a lot but I would welcome examples and templates for branching out a bit more.
* **Jupyter cell output**: nbsphinx has code cell outputs that look very similar to the jupyter notebook style, for better or worse. I really love the features JupyterBook provides with collapsing output/input, scrolling, etc. These are awesome! I believe that the background rendering method for JB leads to the output looking less like a notebook but regardless I would like to see examples of modifying the CSS beyond accent colors and similar to previous bullet, more or less show how a different aesthetic could be approached.

### Note to EB team
This is coming from a place of huge respect and enthusiasm for this project. I am a fan and this discussion will hopefully teach me some new things or start a good conversation, but the goal is to improve not complain. The latter two points are really hoping to find new examples of docs that are customizing the style a bit more than I know of or consider working on this for the EB roadmap.

**Comment by michaelweinold** [comment]
I'd be interested in an update as well - it seems that with the reduced maintainership of @chrisjsewell  (cf. https://github.com/executablebooks/meta/issues/961), development has stalled somewhat. Perhaps more importantly, existing pull requests (some aimed at enabling `MyST-Parser>=1.0`) can't be merged without approval from members of [the Executable Books team](https://github.com/orgs/executablebooks/people). We've greatly benefited from the tools provided by the team so far - perhaps there is a way to invite community members to join the development team such that MyST-NB is maintained again.
